cloudbuild.yaml

```diff
@@ -1,12 +1,39 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
 steps:
 - name: 'golang'
-  env: ['GO111MODULE=on']
-  args: ['go', 'build', './...']
+  id: 'Format golang files'
+  args: ['go', 'fmt', './...']
+
 # We need to set the CLOUDBUILD env var as a signal to skip tests that are
 # not passing in Cloud Build yet. See issue #24.
 - name: 'golang'
+  id: 'Run unit tests'
   env: ['GO111MODULE=on', 'CLOUDBUILD=on']
   args: ['go', 'test', '-v', '-cover', './...']
+
 - name: 'gcr.io/cloud-builders/docker'
-  id: Test build of server container
-  args: ['build', '-t', 'config-validator', '.']
\ No newline at end of file
+  id: 'Build proto-builder'
+  args: ['build', '-t', 'gcv-proto-builder', '-f', './build/proto/Dockerfile', '.']
+
+- name: 'gcr.io/cloud-builders/docker'
+  id: 'Create proto files'
+  args: ['run', '-v', '/workspace:/go/src/github.com/forseti-security/config-validator', 'gcv-proto-builder', 'protoc', '-I/proto', '-I./api', '--go_out=plugins=grpc:./pkg/api/validator', './api/validator.proto']
+
+- name: 'gcr.io/cloud-builders/docker'
+  id: Build config validator container
+  args: ['build', '-t', 'gcr.io/$PROJECT_ID/config-validator:latest', '.']
+
+images: ['gcr.io/$PROJECT_ID/config-validator:latest']
```

Comment:
@kevensen Since this will actually *publish* the container, we only want it running on master.

Could you provide a separate `cloudbuild.yaml` for testing PRs (ie. that doesn't publish the image)?

Reply:
sure thing

Reply:
Can we put them into a `build/` or `cloudbuild/` directory?

Naming:
- `test-cloudbuild.yaml`
- `build-cloudbuild.yaml`

Reply:
done!